src/main/java/org/eclipse/tractusx/puris/backend/stock/controller/StockController.java

```diff
@@ -30,25 +37,31 @@
 import org.eclipse.tractusx.puris.backend.stock.domain.model.MaterialStock;
 import org.eclipse.tractusx.puris.backend.stock.domain.model.PartnerProductStock;
 import org.eclipse.tractusx.puris.backend.stock.domain.model.ProductStock;
-import org.eclipse.tractusx.puris.backend.stock.logic.dto.MaterialStockDto;
-import org.eclipse.tractusx.puris.backend.stock.logic.dto.PartnerProductStockDto;
-import org.eclipse.tractusx.puris.backend.stock.logic.dto.ProductStockDto;
+import org.eclipse.tractusx.puris.backend.stock.logic.dto.*;
 import org.eclipse.tractusx.puris.backend.stock.logic.service.MaterialStockService;
 import org.eclipse.tractusx.puris.backend.stock.logic.service.PartnerProductStockService;
 import org.eclipse.tractusx.puris.backend.stock.logic.service.ProductStockService;
 import org.modelmapper.ModelMapper;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
 import org.springframework.web.bind.annotation.*;
 
-import java.util.Date;
-import java.util.List;
-import java.util.UUID;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import java.util.*;
 import java.util.stream.Collectors;
 
 @RestController
 @RequestMapping("stockView")
+@Slf4j
 public class StockController {
 
+    @Value("${edc.dataplane.public.port}")
+    private String dataPlanePort;
```

Comment:
These information should be obtained from the EndpointDataReference token - together with the authCode.

Reply:
As long as the puris backend is existing outside of the kubernetetes cluster (which is the case in our current development environment), the endpoint information received from the counterparty's unfortunately is not usable, since it is a dns address that only works within the cluster. 

However I have now created a flag in the properties, where the user can decide whether to use the endpoint he received in the message or an address that is modified to work under our current conditions.